commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/VarianceTest.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "org.apache.commons.statistics.descriptive.TestData#testCombine")
+    void testArrayOfArrays(double[] array1, double[] array2) {
+        final double[] combinedArray = TestHelper.concatenate(array1, array2);
+        final double expected = computeExpectedVariance(combinedArray);
+        final double[][] values = {array1, array2};
+        final double actual = Arrays.stream(values)
+                .map(Variance::of)
+                .reduce(Variance::combine)
+                .map(Variance::getAsDouble)
+                .orElseThrow(RuntimeException::new);
+        TestHelper.assertEquals(expected, actual, ULP_ARRAY_OF_ARRAYS, () -> "array of arrays combined variance");
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = {"org.apache.commons.statistics.descriptive.TestData#testCombineNonFinite", "testCombineVarianceNonFinite"})
+    void testCombineNonFinite(double[][] values, double expected) {
+        Variance var1 = Variance.create();
+        Variance var2 = Variance.create();
+        Arrays.stream(values[0]).forEach(var1);
+        Arrays.stream(values[1]).forEach(var2);
+        final double mean2BeforeCombine = var2.getAsDouble();
+        var1.combine(var2);
+        Assertions.assertEquals(expected, var1.getAsDouble(), "combine non-finite");
+        Assertions.assertEquals(mean2BeforeCombine, var2.getAsDouble());
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = {"org.apache.commons.statistics.descriptive.TestData#testCombineNonFinite", "testCombineVarianceNonFinite"})
+    void testCombineRandomOrderNonFinite(double[][] values, double expected) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        final double[] data = TestHelper.concatenate(values[0], values[1]);
+        final int n = values[0].length;
+        for (int i = 1; i <= 10; i++) {
+            for (int j = 1; j <= 10; j++) {
+                TestHelper.shuffle(rng, values[0]);
+                TestHelper.shuffle(rng, values[1]);
+                testCombineNonFinite(values, expected);
+            }
+            TestHelper.shuffle(rng, data);
+            System.arraycopy(data, 0, values[0], 0, n);
+            System.arraycopy(data, n, values[1], 0, values[1].length);
+            testCombineNonFinite(values, expected);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = {"org.apache.commons.statistics.descriptive.TestData#testCombineNonFinite", "testCombineVarianceNonFinite"})
+    void testArrayOfArraysNonFinite(double[][] values, double expected) {
+        final double actual = Arrays.stream(values)
+                .map(Variance::of)
+                .reduce(Variance::combine)
+                .map(Variance::getAsDouble)
+                .orElseThrow(RuntimeException::new);
+        Assertions.assertEquals(expected, actual, "array of arrays combined variance non-finite");
+    }
+
+    static Stream<Arguments> testCombineVarianceNonFinite() {
```

Comment:
Again, these are not the same as the MeanTest. So we are missing cases in one or the other. Please consolidate them to TestData.